songhyun/app.js

```diff
@@ -42,6 +44,137 @@ app.post("/join", async (req, res) => {
     [name, password, email]
   );
   res.status(201).json({ message: "user created" });
+
+// 글 등록(과제3)
+app.post("/post", async (req, res) => {
+  const { title, content, user_id } = req.body;
+
+  await myDataSource.query(
+    `INSERT INTO posts(
+      title,
+      content,
+      user_id
+    ) VALUES (?, ?, ?);
+    `,
+    [title, content, user_id]
+  );
+  res.status(201).json({ message: "postCreated" });
+});
+
+// 글 보기(과제4)
+app.get("/post/view", async (req, res) => {
+  await myDataSource.query(
+    `SELECT
+      posts.user_id AS userId,
+      posts.id AS postingId,
+      posts.imageUrl AS postingImageUrl,
+      posts.content AS postingContent,
+      users.userProfileImage
+    FROM posts
+    INNER JOIN users ON posts.user_id = users.id`,
+    (err, rows) => {
+      res.status(200).json({ data: rows });
+    }
+  );
+});
+
+// 유저가 작성한 글 보기(과제5)
+app.get("/user/view", async (req, res) => {
+  try {
+    const { user_id } = req.body;
+    const userInfo = await myDataSource.query(
+      `SELECT
+        users.id AS userId,
+        users.userProfileImage
+      FROM users
+      WHERE users.id=${user_id}`
+    );
+
+    const posting = await myDataSource.query(
+      `SELECT
+        posts.id AS postingId,
+        posts.imageUrl AS postingImageUrl,
+        posts.content AS postingContent
+      FROM posts, users
+      WHERE posts.user_id=users.id`
+    );
+    userInfo[0]["postings"] = posting;
+    res.status(200).json({ data: userInfo });
+  } catch (err) {
+    console.log("유저가 존재하지 않습니다.");
+  }
+});
+
+// 특정 글 내용 수정(과제6)
+app.patch("/post/update", async (req, res) => {
+  const { updatedContent } = req.body;
+
+  await myDataSource.query(
+    `UPDATE posts
+      SET content="${updatedContent}"
+      WHERE posts.id=1`
```

Comment:
```suggestion
      WHERE posts.id=1`
```
id가 하드 코딩되어 있는 것 같네요.